carp.deployment.core/src/commonMain/kotlin/dk/cachet/carp/deployment/application/DeploymentServiceHost.kt

```diff
@@ -71,8 +71,20 @@ class DeploymentServiceHost( private val repository: DeploymentRepository, priva
 
         val device = deployment.registrableDevices.firstOrNull { it.device.roleName == deviceRoleName }
             ?: throw IllegalArgumentException( "The specified device role name could not be found in the study deployment." )
-        deployment.registerDevice( device.device, registration )
 
+        // Early out when the device is already registered.
+        val priorRegistration = deployment.registeredDevices[ device.device ]
+        if ( priorRegistration == registration )
```

Comment:
You could remove the `if` statement by using `require` with an exception message.

Reply:
Good point! `else require ( priorRegistration == null ) { "error" }`